components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/processors/RandomPasswordProcessor.java

```diff
@@ -0,0 +1,272 @@
+            }
+        }
+        return randomPasswordArrayList.toArray(new RandomPassword[randomPasswordArrayList.size()]);
+    }
+
+    /**
+     * Remove random password list with corresponding old original password if value has not been changed
+     * @param properties
+     * @param randomPasswords
+     */
+    private void replaceRandomPasswordsWithOriginalPasswords(Property[] properties, RandomPassword randomPasswords[]) {
+
+        if (ArrayUtils.isEmpty(properties) || ArrayUtils.isEmpty(randomPasswords)) {
+            return;
+        }
+
+        for (Property property : properties) {
+            if (property == null || StringUtils.isBlank(property.getName()) ||
+                StringUtils.isBlank(property.getValue())) {
+                continue;
+            }
+            if (property.getName().contains(IdentityApplicationConstants.PASSWORD)) {
+                for (RandomPassword randomPassword : randomPasswords) {
+                    if (property.getName().equals(randomPassword.getPropertyName())) {
+                        //if password property value equal to random phrase, user didn't change the password hence set
+                        // old password
+                        if (property.getValue().equals(randomPassword.getRandomPhrase())) {
+                            if (log.isDebugEnabled()){
+                                log.debug("User didn't changed password property: " + property.getName() +
+                                          " hence replace Random Password with Original Password");
+                            }
+                            property.setValue(randomPassword.getPassword());
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private Property[] addUniqueIdProperty(Property [] properties) {
+
+        if (ArrayUtils.isEmpty(properties)){
+            return new Property[0];
+        }
+
+        String uuid = UUID.randomUUID().toString();
+        Property uniqueIdProperty = new Property();
+        uniqueIdProperty.setName(IdentityApplicationConstants.UNIQUE_ID_CONSTANT);
+        uniqueIdProperty.setValue(uuid);
+        if (log.isDebugEnabled()){
+            log.debug("Adding uniqueId property: " + uuid);
+        }
+        properties = (Property[]) ArrayUtils.add(properties, uniqueIdProperty);
+
+        return properties;
+    }
+
+    private Property[] removeUniqueIdProperty(Property [] properties) {
+
+        if (ArrayUtils.isEmpty(properties)){
+            return new Property[0];
+        }
+
+        for (int i=0 ; i < properties.length; i++) {
+            if (properties[i] == null){
+                continue;
+            }
+            if (IdentityApplicationConstants.UNIQUE_ID_CONSTANT.equals(properties[i].getName())) {
+                Property[] propertiesTemp = properties;
+
+                if (log.isDebugEnabled()){
+                    log.debug("Removing uniqueId property: " + properties[i].getName());
```

Comment:
Use StringBuilder and build the entire string and print, otherwise getting mess when printing in multiple threads